src/citrine/resources/design_execution.py

```diff
@@ -0,0 +1,191 @@
+        properties.List(properties.String()),
+        'status_info',
+        serializable=False
+    )
+    experimental = properties.Boolean("experimental", serializable=False, default=True)
+    experimental_reasons = properties.Optional(
+        properties.List(properties.String()),
+        'experimental_reasons',
+        serializable=False
+    )
+
+    def __init__(self):
+        """This shouldn't be called, but it defines members that are set elsewhere."""
+        self.project_id: Optional[UUID] = None  # pragma: no cover
+        self.session: Optional[Session] = None  # pragma: no cover
+
+    def __str__(self):
+        return '<DesignExecution {!r}>'.format(str(self.uid))
+
+    def _path(self):
+        return '/projects/{project_id}/design-workflows/{workflow_id}/executions/{execution_id}' \
+            .format(project_id=self.project_id,
+                    workflow_id=self.workflow_id,
+                    execution_id=self.uid)
+
+    @lru_cache()
+    def results(self,
+                page: Optional[int] = None,
+                per_page: int = 100,
+                ) -> Iterable[DesignCandidate]:
+        """Fetch the Design Candidates for the particular execution, paginated."""
+        path = self._path() + '/candidates'
+
+        fetcher = partial(self._fetch_page, path=path)
+
+        return self._paginator.paginate(page_fetcher=fetcher,
+                                        collection_builder=self._build_candidates,
+                                        page=page,
+                                        per_page=per_page)
+
+
+class DesignExecutionCollection(Collection["DesignExecution"]):
+    """A collection of DesignExecutions."""
+
+    _path_template = '/projects/{project_id}/design-workflows/{workflow_id}/executions'  # noqa
+    _individual_key = None
+    _collection_key = 'response'
+    _resource = DesignExecution
+
+    def __init__(self, *,
+                 project_id: UUID,
+                 session: Session,
+                 workflow_id: Optional[UUID] = None):
+        self.project_id: UUID = project_id
+        self.session: Session = session
+        self.workflow_id: UUID = workflow_id
+
+    def build(self, data: dict) -> DesignExecution:
+        """Build an individual DesignWorkflowExecution."""
+        execution = DesignExecution.build(data)
+        execution.session = self.session
+        execution.project_id = self.project_id
+        return execution
+
+    def trigger(self, execution_input: Score):
+        """Trigger a Design Workflow execution given a score."""
+        path = self._get_path()
+        data = self.session.post_resource(path, {'score': execution_input.dump()})
+        self._check_experimental(data)
+        return self.build(data)
+
+    def _build_candidates(self, subset_collection: Iterable[dict]) -> Iterable[DesignCandidate]:
+        for candidate in subset_collection:
+            yield DesignCandidate.build(candidate)
+
+    def candidates(self,
```

Comment:
Does this duplicate the functionality of DesignWorkflow.results?

Reply:
removed